app/services/builders/metric_chart/base.rb

```diff
@@ -1,26 +1,26 @@
 module Builders
-  module BlogMetricChart
+  module MetricChart
     class Base < BaseService
       def initialize(months = 13)
         @months = months
       end
 
       def call
-        datasets = Technology.all.map do |technology|
+        datasets = entity_type.all.map do |entity|
           generate_results_for(
-            entity_name: technology.name.titlecase,
-            metrics: technology.metrics,
+            entity_name: entity_name(entity),
```

Comment:
when a base class requires the actual implementations to define certain methods, I like to make it explicit like

```ruby
        def entity_name(_technology)
          raise NotImplementedError
        end
```

But I know my friend @juanmanuelramallo don't like this pattern. So feel free to ignore it 😄

Reply:
It's funny because I used to do what you say and then we discussed this with him [here](https://github.com/rootstrap/best_buy_ruby/pull/8#discussion_r393093188).
I do agree with keeping it documented in some way. Even better if it's in a _runnable_ way (not just comments).
Maybe we can create some sort of custom exception for these cases? Or we could just bring the discussion to #ruby and fight to death (?

Reply:
hahaha I no longer have a strong opinion on this. As long as we follow this approach everywhere in the codebase I'm ok with it 👌